packages/headless/src/database/mongodb/wishList/index.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@nestjs/common';
+import { Product } from 'src/entity/product';
+import { WishList, WishlistItem } from 'src/entity/wishList';
+import { IWishListDatabase } from 'src/modules/wishlist/repositories/wishList.database.interface';
+import { ProductModel } from '../product/product.model';
+import { WishListModel } from './wishList.model';
+@Injectable()
+export class WishListDatabase implements IWishListDatabase {
+
+  async getWishlistProduct(wishlist: WishList): Promise<WishList | null> {
+    const products: Product[] = await ProductModel.find({ id: { $in: wishlist.items.map(item => item.productId) } }).select('info photos id -_id');
+
+    let map = new Map<string, Product>();
+    for (let i = 0, len = products.length; i < len; i++) {
+      map.set(products[i].id, products[i]);
+    }
+
+    return {
+      ...wishlist, items: wishlist.items.map(item => { return { ...item, product: map.get(item.productId) } })
+    }
+  }
+
+  async incrementItemQuantity(userId: string, item: WishlistItem,): Promise<WishList | null> {
+    return await WishListModel.findOneAndUpdate(
+      {
+        userId,
+        'items.productId': item.productId,
+      },
+      { $inc: { 'items.$.quantity': item.quantity } },
+      { new: true },
+    )
+      .lean()
+      .exec();
+  }
+
+  async addWishListItem(userId: string, item: WishlistItem,): Promise<WishList | null> {
+    return await WishListModel.findOneAndUpdate(
+      { userId },
+      { $push: { items: item } },
+      { new: true },
+    )
+      .lean()
+      .exec();
+  }
+
+  async createWishList(wishlist: WishList): Promise<WishList | null> {
+    const createdWishlist = await WishListModel.create(wishlist);
+    return createdWishlist.toJSON();
```

Comment:
use "toObject" instead of "toJSON". toJSON converts the full document in json string, so the actual types are being removed.